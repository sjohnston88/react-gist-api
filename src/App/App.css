
.GistPreview {
  padding-left:19%;
  width:80%;
}
