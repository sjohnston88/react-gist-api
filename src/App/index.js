import React, {Component} from 'react';
import queryString from 'query-string';
import moment from 'moment';
import axios from 'axios';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';
import SideBar from '../SideBar/index';
import MainContent from '../MainContent/index'

import './index.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      dataFound: true,
      open: false,
      siteName: 'GitHub Gists',
      invalidDate: false,
      userName: '',
      gistPreview: '',
      gistData: []
    }
    
    this.handleMenuIcon = this.handleMenuIcon.bind(this);
    this.handleOverlayClick = this.handleOverlayClick.bind(this);
    this.handleClick = this.handleClick.bind(this);
    this.getPublicGists = this.getPublicGists.bind(this);
  }

  handleClick(id, e){
    e.preventDefault();
    this.setState({ 
      open: false,
      gistPreview: id
    })
    window.scrollTo(0, 0)
  }
  
  handleMenuIcon(e, open){
    this.setState({ 
      open: !open,
    })
  }
  
  handleOverlayClick(e, open){
    this.setState({ 
      open: !open,
    })
  }

  async getPublicGists(userName, since) {
    try {
      let gistData = await axios.get(`https://api.github.com/users/${userName}/gists${since ? `?since=${since}` : ''}`);
      return gistData;
    } catch (error) {
      //console.error(error);
    }
  }

  async componentDidMount(){ 
    const userName = this.props.match.params.username;
    let since;
    if (this.props.location.search){
      let parsedQuery = queryString.parse(this.props.location.search);
      if (parsedQuery.since){
        if (moment(parsedQuery.since).isValid()){
          since = moment(parsedQuery.since).toISOString();
        } else {
          this.setState({ 
            invalidDate: true,
          })
        }
      }
    }
    let gistData = await this.getPublicGists(userName, since)
    if(gistData && gistData.data.length){
      this.setState({
        userName: userName,
        gistPreview: gistData.data[0].id,
        gistData: gistData.data
      });
    } else {
      this.setState({
        dataFound: false
      });
    }
  }
  
  render() {
    return (
      <MuiThemeProvider>
        <div>
          <AppBar title={this.state.siteName} onLeftIconButtonClick={this.handleMenuIcon} style={{ position: "fixed" }} />
          <SideBar 
            open={this.state.open} 
            title={this.state.siteName} 
            dataFound={this.state.dataFound} 
            gistData={this.state.gistData} 
            onRequestChange={this.handleOverlayClick} 
            onClick={this.handleClick}
          />
          <MainContent 
            userName={this.props.match.params.username}
            dataFound={this.state.dataFound}
            gistPreview={this.state.gistPreview}
            invalidDate={this.state.invalidDate}
          />
        </div>
      </MuiThemeProvider>
    );
  }
}

export default App;
