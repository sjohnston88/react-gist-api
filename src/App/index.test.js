import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Component/DefaultMessage', () => {
  const setup = () => shallow(<App 
    location={{search:""}} 
    match={{params: {username: "defunkt"}, isExact: true, path: "/defunkt", url: "/defunkt"}} 
  />);

  it('renders correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot('default');
  });

});
