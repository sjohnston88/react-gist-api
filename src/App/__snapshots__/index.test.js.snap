// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    location={
        Object {
            "search": "",
          }
    }
    match={
        Object {
            "isExact": true,
            "params": Object {
              "username": "defunkt",
            },
            "path": "/defunkt",
            "url": "/defunkt",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <AppBar
                onLeftIconButtonClick={[Function]}
                showMenuIconButton={true}
                style={
                        Object {
                                "position": "fixed",
                              }
                }
                title="GitHub Gists"
                zDepth={1}
        />
        <SideBar
                dataFound={true}
                gistData={Array []}
                onClick={[Function]}
                onRequestChange={[Function]}
                open={false}
                title="GitHub Gists"
        />
        <MainContent
                dataFound={true}
                gistPreview=""
                invalidDate={false}
                userName="defunkt"
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <AppBar
            onLeftIconButtonClick={[Function]}
            showMenuIconButton={true}
            style={
                        Object {
                                    "position": "fixed",
                                  }
            }
            title="GitHub Gists"
            zDepth={1}
/>,
          <SideBar
            dataFound={true}
            gistData={Array []}
            onClick={[Function]}
            onRequestChange={[Function]}
            open={false}
            title="GitHub Gists"
/>,
          <MainContent
            dataFound={true}
            gistPreview=""
            invalidDate={false}
            userName="defunkt"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onLeftIconButtonClick": [Function],
            "showMenuIconButton": true,
            "style": Object {
              "position": "fixed",
            },
            "title": "GitHub Gists",
            "zDepth": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dataFound": true,
            "gistData": Array [],
            "onClick": [Function],
            "onRequestChange": [Function],
            "open": false,
            "title": "GitHub Gists",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dataFound": true,
            "gistPreview": "",
            "invalidDate": false,
            "userName": "defunkt",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <AppBar
                    onLeftIconButtonClick={[Function]}
                    showMenuIconButton={true}
                    style={
                              Object {
                                        "position": "fixed",
                                      }
                    }
                    title="GitHub Gists"
                    zDepth={1}
          />
          <SideBar
                    dataFound={true}
                    gistData={Array []}
                    onClick={[Function]}
                    onRequestChange={[Function]}
                    open={false}
                    title="GitHub Gists"
          />
          <MainContent
                    dataFound={true}
                    gistPreview=""
                    invalidDate={false}
                    userName="defunkt"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <AppBar
              onLeftIconButtonClick={[Function]}
              showMenuIconButton={true}
              style={
                            Object {
                                          "position": "fixed",
                                        }
              }
              title="GitHub Gists"
              zDepth={1}
/>,
            <SideBar
              dataFound={true}
              gistData={Array []}
              onClick={[Function]}
              onRequestChange={[Function]}
              open={false}
              title="GitHub Gists"
/>,
            <MainContent
              dataFound={true}
              gistPreview=""
              invalidDate={false}
              userName="defunkt"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onLeftIconButtonClick": [Function],
              "showMenuIconButton": true,
              "style": Object {
                "position": "fixed",
              },
              "title": "GitHub Gists",
              "zDepth": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "dataFound": true,
              "gistData": Array [],
              "onClick": [Function],
              "onRequestChange": [Function],
              "open": false,
              "title": "GitHub Gists",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "dataFound": true,
              "gistPreview": "",
              "invalidDate": false,
              "userName": "defunkt",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
